exp/capacity-planning.md

```diff
@@ -0,0 +1,96 @@
+
+- Application resources - How much CPU, memory and disk space does your application need? Will the application use hardware- or software-based video encoding, and if it uses hardware-based encoding, how much GPU capacity is needed?
+- Over-commitment - Does every container need dedicated access to the CPU and memory, or can the capacity be shared between several containers?
+- Application requirements - What type of application are you running? What frame rate and what resolution does your application need? How many containers will be running simultaneously? What would be the impact of not being able to serve all users?
+
+See the following sections for detailed information on these aspects.
+
+## Application resources
+
+Depending on the resources that your application requires, choose a suitable [instance type](https://discourse.ubuntu.com/t/instance-types/17764).
+
+The instance type specifies the resources that are available to the application:
+
+- The number of vCPU cores
+- The amount of RAM
+- The amount of disk space
+- The number of GPU slots
+
+If none of the provided instance types fits for your application, you can also manually [configure the resources](https://discourse.ubuntu.com/t/how-to-configure-available-resources/24960) according to your requirements.
+
+When a container for an application is launched, it takes the specified amount of resources. For example, if an application uses the `a2.3` instance type, it requires 2 CPU cores, 3 GB of memory, 3 GB of disk space and no GPU slot. AMS internally summarises the amount of resources used by containers on a single machine and disallows launching additional containers when all resources are used (see [Over-committing resources](#overcommitting) for how to allow a higher resource usage).
+
+<a name="gpu-slots"></a>
+### GPU slots
+
+An additional aspect to take into account when planning your resources is the number of required GPU slots (see [About GPU support](https://discourse.ubuntu.com/t/gpu-support/17768) for more information).
+
+GPUs have limited capacity that can be shared amongst multiple containers, and GPU slots are a way to fine-tune how many containers can run on a given node. In a cluster setup, you define the number of available GPU slots for each node (see [Configure GPU slots](tbd#configure-gpu-slots) for instructions).
+
+To determine the best number of GPU slots for a specific GPU model, consider the following aspects:
+
+- The memory that the GPU provides
+- The memory that a container uses
+- The number of parallel encoding pipelines that the GPU offers
+
+When you launch a container for an application, it reserves the number of GPU slots defined for the application on the node where it is launched. These GPU slots are marked as unavailable until the container is terminated. If no GPU slots are available on the node, containers that require a GPU will not be launched on it. Containers that don't require a GPU can still be launched.
```

Comment:
No, we actually free up resources of containers which are not running. So if a container is in `error` have released all resources but disk space (as disk space can really only be released when the container is removed from disk).

Reply:
Thanks for the correction. 
It's the unreleased disk space rather than GPU slots that could cause a new container to fail in launching from AMS if we keep a number of the containers with `error` state in the AMS.

Reply:
When you launch a container for an application, it reserves ...
->
When you launch a container for an application, AMS reserves ...

The application doesn't reserve the GPU slots for itself on the node where it is launched but AMS does. The latter feels more reasonable.

Reply:
Thanks - updated.